packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit-drilldown.vue

```diff
@@ -50,83 +50,47 @@
 				:options="outputs"
 				is-selectable
 			>
-				<section class="right-side">
-					<Button
-						class="mr-3"
-						outlined
-						severity="secondary"
-						:disabled="isSaveDisabled"
-						label="Save for reuse"
-						@click="showSaveModelModal = true"
-					/>
-				</section>
 				<tera-notebook-error
 					v-if="executeResponse.status === OperatorStatus.ERROR"
 					:name="executeResponse.name"
 					:value="executeResponse.value"
 					:traceback="executeResponse.traceback"
 				/>
-				<template v-else-if="amr">
-					<tera-model-diagram :model="amr" />
-					<tera-model-parts :model="amr" :feature-config="{ isPreview: true }" />
-				</template>
+				<tera-model v-else-if="amr" is-workflow :assetId="amr.id" @on-save="updateNode" />
```

Comment:
Hmm, I didn't realize you are sticking "tera-model" directly in here. I think the `@on-save` might be ambiguous whether it is:
- updating existing model, or
- create a new model
because they are both calling the same callbacks in `tera-save-asset-modal`

```
  // Save method
  if (props.isUpdatingAsset) {
    saveAssetService.updateAddToProject(newAsset, props.assetType, onSave);
  } else {
    saveAssetService.saveAs(newAsset, props.assetType, props.openOnSave, onSave);
  }
```

where "onSave" is

```
function onSave(data: any) {
  emit('on-save', data);
}
```

I wonder if we need to split into { save, save-as ) so the signals are not crossed at the receiver end. It isn't clear if updateNode here is for "update" or "save a new model".

Reply:
I agree with you, but this should be done in another PR that I will do after this one, as this is urgently needed.

Reply:
https://github.com/DARPA-ASKEM/terarium/issues/5021